frame/balances/src/lib.rs

```diff
@@ -826,48 +831,52 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 	/// Update the account entry for `who`, given the locks.
 	fn update_locks(who: &T::AccountId, locks: &[BalanceLock<T::Balance>]) {
-		if locks.len() as u32 > T::MaxLocks::get() {
-			log::warn!(
-				target: "runtime::balances",
-				"Warning: A user has more currency locks than expected. \
-				A runtime configuration adjustment may be needed."
-			);
-		}
-		// No way this can fail since we do not alter the existential balances.
-		let res = Self::mutate_account(who, |b| {
-			b.misc_frozen = Zero::zero();
-			b.fee_frozen = Zero::zero();
-			for l in locks.iter() {
-				if l.reasons == Reasons::All || l.reasons == Reasons::Misc {
-					b.misc_frozen = b.misc_frozen.max(l.amount);
+		unsafe {
+			let bounded_locks = BoundedVec::<_, T::MaxLocks>::force_from(locks.to_vec(), Some("Balances Update Locks"));
```

Comment:
It's legal to more tightly constrain the scope of the `unsafe`, which makes it more obvious which operations were dangerous.

```suggestion
		let bounded_locks = unsafe { BoundedVec::<_, T::MaxLocks>::force_from(locks.to_vec(), Some("Balances Update Locks")) };
```